src/newgrf_config.cpp

```diff
@@ -29,6 +29,7 @@
 
 #include "safeguards.h"
 
+static_assert(MAX_GRF_COUNT >= NETWORK_MAX_GRF_COUNT + NUM_BASESET_NEWGRFS);
```

Comment:
These asserts I always consider "annoying future-us" :) Mainly, as I think it is not needed.

The `MAX_GRF_COUNT` is flexible, and can be any value. So we can just use the other constants to set a valid value for it. I would define it something like:

`static const uint MAX_GRF_COUNT =NETWORK_MAX_GRF_COUNT + NUM_BASESET_NEWGRFS + MAX_STATIC_GRF_COUNT;`

Where the last const is something like 16 or what-ever .. just some safe value. That means that nobody ever again has to worry about this value :)

Reply:
Except that introducing a `MAX_STATIC_GRF_COUNT` implies that you should also check for that limit, which is currently not done. Also 16 might be too limiting as currently it is effectively 126.

Reply:
I was using just any random value :) If you want it bigger, go for it. No need to nitpick like that ;)

And yeah, checking on it would be nice .. but if you don't, it isn't a regression, as we currently also don't check for it. It just explains why `NETWORK_MAX_GRF_COUNT` is 62, and `MAX_GRF_COUNT` is 128. It is just making more visible where those numbers are deduced from.

Reply:
Better idea was to just abolish that completely, i.e. #9431.